cognite/extractorutils/threading.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import signal
+from threading import Condition
+from typing import Any, Optional
+
+
+class CancellationToken:
+    """
+    Abstraction for a hierarchical cancellation token.
+
+    Using this you can create hierarchies of cancellation tokens, to cancel a part of the extractor
+    without cancelling the whole process. Use `create_child_token` to create a token that will be
```

Comment:
Kind of nit, but python doc strings are restructured text and not markdown, so code snippets are double backticked
```suggestion
    without cancelling the whole process. Use ``create_child_token`` to create a token that will be
```